discovery-infra/tests/test_cancel_and_reset.py

```diff
@@ -0,0 +1,115 @@
+import os
+import pytest
+
+import assisted_service_api
+import utils
+import consts
+
+if os.environ.get('NODE_ENV') == 'QE_VM':
+    import node_controlers.qe_vm_controler
+
+ssh_public_key = os.environ.get('SSH_PUB_KEY')
+remote_service_url = os.environ.get('REMOTE_SERVICE_URL')
+pull_secret = os.environ.get('PULL_SECRET')
+cluster_name = os.environ.get('CLUSTER_NAME')
+offline_token = os.environ.get('OFFLINE_TOKEN')
+openshift_version = os.environ.get('OPENSHIFT_VERSION')
+base_domain = os.environ.get('BASE_DOMAIN')
+iso_download_path = os.environ.get('ISO')
+num_masters = int(os.environ.get('NUM_MASTERS'))
+num_workers = int(os.environ.get('NUM_WORKERS'))
+
+nodes_count = num_masters + num_workers
+
+# api_client = assisted_service_api.create_client(remote_service_url)
+# controler = node_controlers.qe_vm_controler.create_controler()
+from tests.base_cancel_reset import BaseCancelReset
+
+class TestCancelReset(BaseCancelReset):
```

Comment:
1. Split into helper functions for readability.
2. I don't see a reason to create a class if you don't hold members or utilizes class operations.

pseudo:

```python
def test_reboot():
    cluster = _new_cluster()
    _generate_and_download_iso(cluster)
    ....

@pytest.fixture()
def _new_cluster()
    return api_client.create_cluster(
            cluster_name, 
            ssh_public_key=ssh_public_key, 
            pull_secret=pull_secret, 
            openshift_version=openshift_version, 
            base_dns_domain=base_domain, 
            vip_dhcp_allocation=False
        )

@pytest.fixture()
def _generate_and_download_iso(cluster):
    api_client.generate_and_download_image(
        cluster_id=cluster.id, 
        ssh_key=ssh_public_key, 
        image_path=iso_download_path
    )
```

Reply:
moved to helper functions in BaseTest class